src/common/vector/value_vector.cpp

```diff
@@ -70,6 +70,33 @@ void ValueVector::copyFromRowData(uint32_t pos, const uint8_t* rowData) {
     }
 }
 
+void ValueVector::copyToRowData(
+    uint32_t pos, uint8_t* rowData, InMemOverflowBuffer* rowOverflowBuffer) const {
+    switch (dataType.getPhysicalType()) {
+    case PhysicalTypeID::STRUCT: {
+        StructVector::copyToRowData(this, pos, rowData, rowOverflowBuffer);
+    } break;
+    case PhysicalTypeID::VAR_LIST: {
+        ListVector::copyToRowData(this, pos, rowData, rowOverflowBuffer);
+    } break;
+    case PhysicalTypeID::STRING: {
+        auto& srcStr = getValue<ku_string_t>(pos);
+        auto& dstStr = *(ku_string_t*)rowData;
+        if (ku_string_t::isShortString(srcStr.len)) {
+            dstStr.setShortString(srcStr);
+        } else {
+            dstStr.overflowPtr =
```

Comment:
I see this `reinterpret_cast<uint64_t>(rowOverflowBuffer->allocateSpace(srcStr.len));` code in multiple places(especially in the string functions).
Maybe we can add a new static function in the StringVector to allocate space for a str based on the strlength?

Reply:
There is a similar issue #1678